<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="b10prot">
<title>b10prot â€¢ b10prot</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="b10prot">
<meta property="og:description" content="b10prot">
<meta property="og:image" content="https://akrogp.github.io/b10prot/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">b10prot</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/b10prot.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>b10prot</h1>
            
      
      
      <div class="d-none name"><code>b10prot.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>This vignette shows how to build different proteomics identification
workflows using the <code>b10prot</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://conflicted.r-lib.org/" class="external-link">conflicted</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://akrogp.github.io/b10prot/">b10prot</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://conflicted.r-lib.org/reference/conflicts_prefer.html" class="external-link">conflicts_prefer</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="va"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="loading-psms">Loading PSMs<a class="anchor" aria-label="anchor" href="#loading-psms"></a>
</h3>
<p>The initial input for the identification workflows presented here
consists of a list of Peptide-to-Spectrum Matches (PSMs) that must have
been previously generated using any search engine. This list of PSMs
must be available in a <code>data.frame</code> containing at least the
following columns:</p>
<ul>
<li><code>psmScore</code></li>
<li><code>rank</code></li>
<li><code>isDecoy</code></li>
<li><code>peptideRef</code></li>
<li><code>proteinRef</code></li>
<li>
<code>geneRef</code> (optional)</li>
</ul>
<p>We provide an <code>iwf_load_psms</code> function to facilitate
obtaining these PSMs from one or more <a href="https://www.psidev.info/mzidentml" class="external-link">mzIdentML</a> (*.mzid) files,
which is a standardized file format by the <a href="https://www.psidev.info/" class="external-link">HUPO Proteomics Standards
Initiative</a>. To parse these mzid files, we rely on the
<code>mzR</code> package.</p>
<p>As an example dataset, we will use three mzid files obtained after
searching three fractions of one tissue from this <a href="https://www.ebi.ac.uk/pride/archive/projects/PXD000561" class="external-link">draft map
of the human proteome</a>. For this search, we used the <a href="https://msgfplus.github.io/" class="external-link">MS-GF+ Search Engine</a> within <a href="https://compomics.github.io/projects/searchgui" class="external-link">SearchGUI</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">psms</span> <span class="op">&lt;-</span> </span>
<span>  <span class="co"># Load PSMs from mzIdentML files</span></span>
<span>  <span class="fu"><a href="../reference/iwf_load_psms.html">iwf_load_psms</a></span><span class="op">(</span></span>
<span>    path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DATA_PATH</span>, <span class="st">"msgf"</span><span class="op">)</span>,</span>
<span>    <span class="co"># We will use MS-GF+ spectral E-value for the target-decoy approach</span></span>
<span>    psm_score <span class="op">=</span> <span class="st">"MS.GF.SpecEValue"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="co"># Extract UniProt accession for convenience (optional)</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>proteinRef <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_split.html" class="external-link">str_split_i</a></span><span class="op">(</span><span class="va">DatabaseAccess</span>, <span class="st">"\\|"</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="co"># Decoy information was not specified in my mzid files</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>isDecoy <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">proteinRef</span>, <span class="st">"_REVERSED"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="co"># Extract UniProt gene name (only if you are interested in this level)</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>geneRef <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_extract.html" class="external-link">str_extract</a></span><span class="op">(</span><span class="va">DatabaseDescription</span>, <span class="st">"GN=\\S+"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>geneRef <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_sub.html" class="external-link">str_sub</a></span><span class="op">(</span><span class="va">geneRef</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>geneRef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">isDecoy</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">geneRef</span>, <span class="st">"_REVERSED"</span><span class="op">)</span>, <span class="va">geneRef</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Only best PSM per spectrum</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">psms</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 49,708</span></span>
<span><span class="co">#&gt; Columns: 29</span></span>
<span><span class="co">#&gt; $ spectrumID               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "index=9901", "index=12199", "index=13496", "â€¦</span></span>
<span><span class="co">#&gt; $ chargeState              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 3, 2, 3, 3, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 3, â€¦</span></span>
<span><span class="co">#&gt; $ rank                     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, â€¦</span></span>
<span><span class="co">#&gt; $ passThreshold            <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUâ€¦</span></span>
<span><span class="co">#&gt; $ experimentalMassToCharge <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 1063.8254, 1141.6216, 994.1609, 1161.4554, 11â€¦</span></span>
<span><span class="co">#&gt; $ calculatedMassToCharge   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 1063.8251, 1141.6218, 994.1621, 1161.4556, 11â€¦</span></span>
<span><span class="co">#&gt; $ sequence                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "EPVSVGTPSEGEGLGADGQEHKEDTFDVFR", "GVVGPGPAALâ€¦</span></span>
<span><span class="co">#&gt; $ peptideRef               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Pep_EPVSVGTPSEGEGLGADGQEHKEDTFDVFR", "Pep_GVâ€¦</span></span>
<span><span class="co">#&gt; $ modNum                   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 1, 6, 1, 0, 0, 0, 3, 1, 1, 0, 0, 0, 1, â€¦</span></span>
<span><span class="co">#&gt; $ isDecoy                  <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALâ€¦</span></span>
<span><span class="co">#&gt; $ post                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Q", "G", "G", "G", "S", "S", "Q", "L", "E", â€¦</span></span>
<span><span class="co">#&gt; $ pre                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "R", "R", "K", "K", "R", "R", "K", "K", "K", â€¦</span></span>
<span><span class="co">#&gt; $ start                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1039, 8, 314, 38, 326, 326, 278, 63, 757, 293â€¦</span></span>
<span><span class="co">#&gt; $ end                      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1068, 36, 342, 64, 350, 350, 303, 91, 785, 31â€¦</span></span>
<span><span class="co">#&gt; $ DatabaseAccess           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "sp|Q8IX01|SUGP2_HUMAN", "sp|P22059|OSBP1_HUMâ€¦</span></span>
<span><span class="co">#&gt; $ DBseqLength              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1082, 807, 561, 478, 353, 353, 589, 142, 851,â€¦</span></span>
<span><span class="co">#&gt; $ DatabaseSeq              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "", "", "", "", "", "", "", "", "", "", "", "â€¦</span></span>
<span><span class="co">#&gt; $ DatabaseDescription      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "sp|Q8IX01|SUGP2_HUMAN SURP and G-patch domaiâ€¦</span></span>
<span><span class="co">#&gt; $ spectrum.title           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Adult_Testis_bRP_Elite_68_f01.12299.12299.3"â€¦</span></span>
<span><span class="co">#&gt; $ scan.number.s.           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 12299, 14751, 16134, 11137, 7177, 8471, 9644,â€¦</span></span>
<span><span class="co">#&gt; $ scan.start.time          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "3695.0874", "4370.828", "4756.3667", "3381.8â€¦</span></span>
<span><span class="co">#&gt; $ acquisitionNum           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 9901, 12199, 13496, 8811, 5099, 6312, 7412, 1â€¦</span></span>
<span><span class="co">#&gt; $ MS.GF.RawScore           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 326, 292, 282, 256, 319, 296, 217, 228, 222, â€¦</span></span>
<span><span class="co">#&gt; $ MS.GF.DeNovoScore        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 331, 297, 288, 261, 319, 296, 219, 232, 232, â€¦</span></span>
<span><span class="co">#&gt; $ MS.GF.SpecEValue         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 1.098553e-35, 3.951045e-35, 4.986274e-35, 2.2â€¦</span></span>
<span><span class="co">#&gt; $ MS.GF.EValue             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 2.428852e-28, 8.729901e-28, 1.101726e-27, 5.0â€¦</span></span>
<span><span class="co">#&gt; $ psmScore                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 1.098553e-35, 3.951045e-35, 4.986274e-35, 2.2â€¦</span></span>
<span><span class="co">#&gt; $ proteinRef               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Q8IX01", "P22059", "P49902", "P04004", "P226â€¦</span></span>
<span><span class="co">#&gt; $ geneRef                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "SUGP2", "OSBP", "NT5C2", "VTN", "HNRNPA2B1",â€¦</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="peptide-identifications">Peptide identifications<a class="anchor" aria-label="anchor" href="#peptide-identifications"></a>
</h3>
<p>From the list of PSMs, we obtain a list of peptides by selecting the
best PSM for each peptide, and then we calculate peptide confidence
scores using the target-decoy approach:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peptides</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">psms</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="co"># Best PSM per peptide</span></span>
<span>  <span class="fu"><a href="../reference/iwf_psm2pep.html">iwf_psm2pep</a></span><span class="op">(</span>lower_better <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Calculate target-decoy approach metrics</span></span>
<span>  <span class="fu"><a href="../reference/target_decoy_approach.html">target_decoy_approach</a></span><span class="op">(</span><span class="va">pepScore</span><span class="op">)</span></span>
<span></span>
<span><span class="va">peptides</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 27,498</span></span>
<span><span class="co">#&gt; Columns: 9</span></span>
<span><span class="co">#&gt; $ peptideRef <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Pep_EPVSVGTPSEGEGLGADGQEHKEDTFDVFR", "Pep_GVVGPGPAALAALGGGâ€¦</span></span>
<span><span class="co">#&gt; $ pepScore   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 1.098553e-35, 1.358987e-35, 4.986274e-35, 2.282557e-34, 2.5â€¦</span></span>
<span><span class="co">#&gt; $ isDecoy    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALâ€¦</span></span>
<span><span class="co">#&gt; $ decoys     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ targets    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 2, 3, 4, 5, 6, 8, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, â€¦</span></span>
<span><span class="co">#&gt; $ pval       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 6.423433e-05, 6.423433e-05, 6.423433e-05, 6.423433e-05, 6.4â€¦</span></span>
<span><span class="co">#&gt; $ LP         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.192233, 4.192233, 4.192233, 4.192233, 4.192233, 4.192233,â€¦</span></span>
<span><span class="co">#&gt; $ FDR        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ qval       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span></code></pre></div>
<p>If we are only interested in peptides, we can obtain the identified
peptides by setting a peptide-level FDR threshold:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peptides</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">qval</span> <span class="op">&lt;=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/global_fdr.html">global_fdr</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 Ã— 3</span></span></span>
<span><span class="co">#&gt;   Target Decoy `Global FDR (%)`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>  <span style="text-decoration: underline;">11</span>626   116            0.998</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="protein-identifications">Protein identifications<a class="anchor" aria-label="anchor" href="#protein-identifications"></a>
</h3>
<p>To obtain a list of protein identifications, we compute a
protein-level score using the scores of the corresponding peptides and
then apply a protein-level FDR. We use LPGF as the protein-level score
using only unique peptides, i.e., peptides not shared by different
proteins.</p>
<p>First, we need a <code>data.frame</code> of peptide-to-protein
relations, including the peptide-level scores from the previous section.
We obtain the peptide-to-protein relations from the initial PSMs and
then merge the peptide-level scores from the peptide list:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep2prot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="co"># Peptide-to-protein relations</span></span>
<span>  <span class="fu"><a href="../reference/iwf_pep2level.html">iwf_pep2level</a></span><span class="op">(</span><span class="va">psms</span>, levelRef <span class="op">=</span> <span class="va">proteinRef</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Include peptide scores</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span><span class="va">peptides</span>, by <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/join_by.html" class="external-link">join_by</a></span><span class="op">(</span><span class="va">peptideRef</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pep2prot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 29,725</span></span>
<span><span class="co">#&gt; Columns: 11</span></span>
<span><span class="co">#&gt; $ peptideRef <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Pep_AAAAAAAAAVSR", "Pep_AAAAAAAKNK", "Pep_AAAAAAALQAK", "Pâ€¦</span></span>
<span><span class="co">#&gt; $ proteinRef <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Q96JP5", "A8MW92", "P36578", "A6NIH7", "Q92859_REVERSED", â€¦</span></span>
<span><span class="co">#&gt; $ shared     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1,â€¦</span></span>
<span><span class="co">#&gt; $ pepScore   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.882813e-17, 8.691400e-08, 9.765627e-16, 9.065173e-19, 4.9â€¦</span></span>
<span><span class="co">#&gt; $ isDecoy    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE, FALSE, FALSE, FALSE, TRUE, TRUE, FALSE, FALSE, FALSEâ€¦</span></span>
<span><span class="co">#&gt; $ decoys     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 424, 0, 0, 7510, 5628, 0, 0, 0, 6948, 0, 7039, 7628, 0, â€¦</span></span>
<span><span class="co">#&gt; $ targets    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 4736, 12041, 5947, 3308, 19427, 17519, 3778, 5003, 2770, 18â€¦</span></span>
<span><span class="co">#&gt; $ pval       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 6.423433e-05, 5.453494e-02, 6.423433e-05, 6.423433e-05, 9.6â€¦</span></span>
<span><span class="co">#&gt; $ LP         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.192232823, 1.263325133, 4.192232823, 4.192232823, 0.01559â€¦</span></span>
<span><span class="co">#&gt; $ FDR        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.00000000, 0.03521302, 0.00000000, 0.00000000, 0.38657538,â€¦</span></span>
<span><span class="co">#&gt; $ qval       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.00000000, 0.03521302, 0.00000000, 0.00000000, 0.38653559,â€¦</span></span></code></pre></div>
<p>Now, we can collapse these relationships into a list of protein
identifications with protein-level scores:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">proteins</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">pep2prot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Only consider unique (i.e. not shared) peptides</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">shared</span><span class="op">==</span><span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Calculate protein-level scores</span></span>
<span>  <span class="fu"><a href="../reference/lpg.html">lpg</a></span><span class="op">(</span><span class="va">proteinRef</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Calculate target-decoy approach metrics using the LPGF score</span></span>
<span>  <span class="fu"><a href="../reference/target_decoy_approach.html">target_decoy_approach</a></span><span class="op">(</span><span class="va">LPGF</span>, lower_better <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 13,579</span></span>
<span><span class="co">#&gt; Columns: 19</span></span>
<span><span class="co">#&gt; $ proteinRef <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "A0AVT1", "A1L0T0", "A2RRP1", "A5YKK6", "A6NHR9", "O00410",â€¦</span></span>
<span><span class="co">#&gt; $ isDecoy    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALâ€¦</span></span>
<span><span class="co">#&gt; $ n          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 10, 7, 7, 6, 7, 11, 8, 8, 8, 8, 8, 9, 6, 10, 8, 10, 7, 8, 7â€¦</span></span>
<span><span class="co">#&gt; $ m          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 9, 6, 7, 6, 6, 11, 6, 6, 8, 7, 8, 8, 6, 6, 8, 10, 6, 7, 6, â€¦</span></span>
<span><span class="co">#&gt; $ LPM        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.192233, 4.192233, 4.192233, 4.192233, 4.192233, 4.192233,â€¦</span></span>
<span><span class="co">#&gt; $ LPS        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 36.99811, 24.35836, 29.34563, 25.15340, 25.29808, 46.11456,â€¦</span></span>
<span><span class="co">#&gt; $ LPF        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 36.05800, 24.30830, 29.34563, 25.15340, 25.15340, 46.11456,â€¦</span></span>
<span><span class="co">#&gt; $ LPGM       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 3.192358, 3.347218, 3.347218, 3.414151, 3.347218, 3.150980,â€¦</span></span>
<span><span class="co">#&gt; $ LPGS       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300,â€¦</span></span>
<span><span class="co">#&gt; $ LPGF       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300,â€¦</span></span>
<span><span class="co">#&gt; $ peptideRef <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Pep_CLANLRPLLDSGTM+16GTK", "Pep_AAM+16GLGAR", "Pep_CSGALTVâ€¦</span></span>
<span><span class="co">#&gt; $ shared     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,â€¦</span></span>
<span><span class="co">#&gt; $ pepScore   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 5.559802e-13, 7.812501e-12, 9.765627e-16, 3.906251e-16, 8.3â€¦</span></span>
<span><span class="co">#&gt; $ decoys     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ targets    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 366, 366, 366, 366, 366, 366, 366, 366, 366, 366, 366, 366,â€¦</span></span>
<span><span class="co">#&gt; $ pval       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 9.425071e-05, 9.425071e-05, 9.425071e-05, 9.425071e-05, 9.4â€¦</span></span>
<span><span class="co">#&gt; $ FDR        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ LP         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.025715, 4.025715, 4.025715, 4.025715, 4.025715, 4.025715,â€¦</span></span>
<span><span class="co">#&gt; $ qval       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span></code></pre></div>
<p>And check whether the distribution of protein-level scores in the
decoy proteins follows a uniform distribution. We can see that the LPGF
scores perform as expected:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/plot_rank.html">plot_rank</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="b10prot_files/figure-html/unnamed-chunk-10-1.png" width="672"></p>
<p>Finally, we can obtain the identified proteins by applying a
protein-level FDR threshold:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">qval</span> <span class="op">&lt;=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/global_fdr.html">global_fdr</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 Ã— 3</span></span></span>
<span><span class="co">#&gt;   Target Decoy `Global FDR (%)`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>   <span style="text-decoration: underline;">4</span>248    42            0.989</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="gene-identifications">Gene identifications<a class="anchor" aria-label="anchor" href="#gene-identifications"></a>
</h3>
<p>Since we have used only unique peptides in the previous section,
protein isoforms from the same gene that share the same peptide
sequences would have been removed. To minimize the effect of this
simplification, we will now consider peptides that are unique at the
gene level. The steps are equivalent to those used in the protein
identification workflow.</p>
<p>First, we need a <code>data.frame</code> of peptide-to-gene
relations, including the peptide-level scores from the previous section.
Once again, we obtain the peptide-to-gene relations from the initial
PSMs and then merge the peptide-level scores from the peptide list:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep2gene</span> <span class="op">&lt;-</span> </span>
<span>  <span class="co"># Peptide-to-gene relations</span></span>
<span>  <span class="fu"><a href="../reference/iwf_pep2level.html">iwf_pep2level</a></span><span class="op">(</span><span class="va">psms</span>, levelRef <span class="op">=</span> <span class="va">geneRef</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Include peptide scores</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span><span class="va">peptides</span>, by <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/join_by.html" class="external-link">join_by</a></span><span class="op">(</span><span class="va">peptideRef</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pep2gene</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 29,703</span></span>
<span><span class="co">#&gt; Columns: 11</span></span>
<span><span class="co">#&gt; $ peptideRef <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Pep_AAAAAAAAAVSR", "Pep_AAAAAAAKNK", "Pep_AAAAAAALQAK", "Pâ€¦</span></span>
<span><span class="co">#&gt; $ geneRef    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "ZFP91", "PHF20L1", "RPL4", "UNC119B", "NEO1_REVERSED", "VAâ€¦</span></span>
<span><span class="co">#&gt; $ shared     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1,â€¦</span></span>
<span><span class="co">#&gt; $ pepScore   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.882813e-17, 8.691400e-08, 9.765627e-16, 9.065173e-19, 4.9â€¦</span></span>
<span><span class="co">#&gt; $ isDecoy    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE, FALSE, FALSE, FALSE, TRUE, TRUE, FALSE, FALSE, FALSEâ€¦</span></span>
<span><span class="co">#&gt; $ decoys     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 424, 0, 0, 7510, 5628, 0, 0, 0, 6948, 0, 7039, 7628, 0, â€¦</span></span>
<span><span class="co">#&gt; $ targets    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 4736, 12041, 5947, 3308, 19427, 17519, 3778, 5003, 2770, 18â€¦</span></span>
<span><span class="co">#&gt; $ pval       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 6.423433e-05, 5.453494e-02, 6.423433e-05, 6.423433e-05, 9.6â€¦</span></span>
<span><span class="co">#&gt; $ LP         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.192232823, 1.263325133, 4.192232823, 4.192232823, 0.01559â€¦</span></span>
<span><span class="co">#&gt; $ FDR        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.00000000, 0.03521302, 0.00000000, 0.00000000, 0.38657538,â€¦</span></span>
<span><span class="co">#&gt; $ qval       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.00000000, 0.03521302, 0.00000000, 0.00000000, 0.38653559,â€¦</span></span></code></pre></div>
<p>Now, we can collapse these relationships into a list of gene
identifications with gene-level scores:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">pep2gene</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Only consider unique (i.e. not shared) peptides</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">shared</span><span class="op">==</span><span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Calculate gene-level scores</span></span>
<span>  <span class="fu"><a href="../reference/lpg.html">lpg</a></span><span class="op">(</span><span class="va">geneRef</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Calculate target-decoy approach metrics using the LPGF score</span></span>
<span>  <span class="fu"><a href="../reference/target_decoy_approach.html">target_decoy_approach</a></span><span class="op">(</span><span class="va">LPGF</span>, lower_better <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 13,552</span></span>
<span><span class="co">#&gt; Columns: 19</span></span>
<span><span class="co">#&gt; $ geneRef    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "A2M", "AARS", "ABAT", "ACAA2", "ACAT1", "ACLY", "ACO1", "Aâ€¦</span></span>
<span><span class="co">#&gt; $ isDecoy    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALâ€¦</span></span>
<span><span class="co">#&gt; $ n          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 12, 9, 8, 10, 9, 10, 8, 11, 11, 6, 6, 8, 6, 10, 9, 69, 7, 7â€¦</span></span>
<span><span class="co">#&gt; $ m          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 12, 9, 6, 10, 9, 9, 7, 11, 10, 6, 6, 7, 6, 8, 9, 61, 7, 7, â€¦</span></span>
<span><span class="co">#&gt; $ LPM        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.192233, 4.192233, 4.192233, 4.192233, 4.192233, 4.192233,â€¦</span></span>
<span><span class="co">#&gt; $ LPS        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 50.30679, 37.73010, 24.49706, 41.92233, 36.88500, 38.09116,â€¦</span></span>
<span><span class="co">#&gt; $ LPF        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 50.30679, 37.73010, 23.92295, 41.92233, 36.88500, 37.73010,â€¦</span></span>
<span><span class="co">#&gt; $ LPGM       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 3.113205, 3.238102, 3.289240, 3.192358, 3.238102, 3.192358,â€¦</span></span>
<span><span class="co">#&gt; $ LPGS       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 300.00000, 300.00000, 15.95459, 300.00000, 300.00000, 300.0â€¦</span></span>
<span><span class="co">#&gt; $ LPGF       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300,â€¦</span></span>
<span><span class="co">#&gt; $ peptideRef <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Pep_DLTGFPGPLNDQDNEDCLNR", "Pep_AVFDETYPDPVR", "Pep_CLEEVEâ€¦</span></span>
<span><span class="co">#&gt; $ shared     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,â€¦</span></span>
<span><span class="co">#&gt; $ pepScore   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 1.521663e-15, 1.464844e-16, 3.906251e-15, 2.723810e-29, 1.8â€¦</span></span>
<span><span class="co">#&gt; $ decoys     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ targets    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 367, 367, 367, 367, 367, 367, 367, 367, 367, 367, 367, 367,â€¦</span></span>
<span><span class="co">#&gt; $ pval       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 9.444654e-05, 9.444654e-05, 9.444654e-05, 9.444654e-05, 9.4â€¦</span></span>
<span><span class="co">#&gt; $ FDR        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ LP         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.024814, 4.024814, 4.024814, 4.024814, 4.024814, 4.024814,â€¦</span></span>
<span><span class="co">#&gt; $ qval       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span></code></pre></div>
<p>And check whether the distribution of gene-level scores in the decoy
genes follows a uniform distribution. We can see that the LPGF scores
perform as expected:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/plot_rank.html">plot_rank</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="b10prot_files/figure-html/unnamed-chunk-14-1.png" width="672"></p>
<p>Finally, we can obtain the identified genes by applying a gene-level
FDR threshold:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">qval</span> <span class="op">&lt;=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/global_fdr.html">global_fdr</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 Ã— 3</span></span></span>
<span><span class="co">#&gt;   Target Decoy `Global FDR (%)`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>   <span style="text-decoration: underline;">4</span>249    42            0.988</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="protein-groups">Protein groups<a class="anchor" aria-label="anchor" href="#protein-groups"></a>
</h3>
<p>Although working at the gene level reduces the number of shared
peptide sequences, we may still remove peptides whose sequences are
shared between proteins from different genes. To avoid this problem, we
can build groups of proteins that share peptide sequences and report a
list of protein groups passing a protein group-level FDR threshold.</p>
<p>To build these protein groups, we use the PAnalyzer algorithm. An
example is included with this package:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">example_panalyzer</span>, package <span class="op">=</span> <span class="st">"b10prot"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_groups.html">plot_groups</a></span><span class="op">(</span><span class="va">example_panalyzer</span>, groupRefs <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<p><img src="b10prot_files/figure-html/unnamed-chunk-16-1.png" width="672"></p>
<p>PAnalyzer receives as input a <code>data.frame</code> with
peptide-to-protein relations and returns another <code>data.frame</code>
that includes peptide-to-protein-to-group relations along with the
corresponding peptide and protein types:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep2prot2group</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">pep2prot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/iwf_grouping.html">iwf_grouping</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pep2prot2group</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 Ã— 5</span></span></span>
<span><span class="co">#&gt;   Type              TargetProteins DecoyProteins TargetGroups DecoyGroups</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ambiguous                     26            <span style="color: #BB0000;">NA</span>            2          <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> conclusive                  <span style="text-decoration: underline;">8</span>155          <span style="text-decoration: underline;">5</span>300         <span style="text-decoration: underline;">8</span>155        <span style="text-decoration: underline;">5</span>300</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> indistinguishable            528           320          203         126</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> non-conclusive               335           142          335         142</span></span></code></pre></div>
<p>From these relations, we can obtain the list of protein group
identifications with their corresponding scores:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">pep2prot2group</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Instead of iwf_pep2level() we use iwf_pep2group() to retain the list of proteins within each group</span></span>
<span>  <span class="fu"><a href="../reference/iwf_pep2group.html">iwf_pep2group</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Only consider peptides unique to one group (this also removes non-conclusice proteins) </span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">shared</span><span class="op">==</span><span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Calculate protein group-level scores</span></span>
<span>  <span class="fu"><a href="../reference/lpg.html">lpg</a></span><span class="op">(</span><span class="va">groupRef</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Calculate target-decoy approach metrics using the LPGF score</span></span>
<span>  <span class="fu"><a href="../reference/target_decoy_approach.html">target_decoy_approach</a></span><span class="op">(</span><span class="va">LPGF</span>, lower_better <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">groups</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">m</span><span class="op">&gt;</span><span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 2,326</span></span>
<span><span class="co">#&gt; Columns: 27</span></span>
<span><span class="co">#&gt; $ groupRef      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 17, 28, 32, 68, 73, 81, 83, 145, 183, 243, 289, 299, 301â€¦</span></span>
<span><span class="co">#&gt; $ isDecoy       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, â€¦</span></span>
<span><span class="co">#&gt; $ n             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 7, 10, 6, 8, 12, 7, 7, 6, 35, 9, 69, 50, 9, 11, 6, 8, 15â€¦</span></span>
<span><span class="co">#&gt; $ m             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 7, 10, 6, 7, 10, 7, 7, 6, 34, 6, 61, 45, 9, 8, 6, 7, 13,â€¦</span></span>
<span><span class="co">#&gt; $ LPM           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.192233, 4.192233, 4.192233, 4.192233, 4.192233, 4.1922â€¦</span></span>
<span><span class="co">#&gt; $ LPS           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 29.34563, 39.00168, 25.15340, 30.15675, 44.63238, 29.345â€¦</span></span>
<span><span class="co">#&gt; $ LPF           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 29.34563, 39.00168, 25.15340, 29.34563, 41.92233, 29.345â€¦</span></span>
<span><span class="co">#&gt; $ LPGM          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 3.347218, 3.192358, 3.414151, 3.289240, 3.113205, 3.3472â€¦</span></span>
<span><span class="co">#&gt; $ LPGS          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 300.00000, 300.00000, 300.00000, 300.00000, 300.00000, 3â€¦</span></span>
<span><span class="co">#&gt; $ LPGF          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 3â€¦</span></span>
<span><span class="co">#&gt; $ peptideRef    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Pep_EVVSLQTSLEQK", "Pep_LDAESLVK", "Pep_EFSLDVGYER", "Pâ€¦</span></span>
<span><span class="co">#&gt; $ proteinRef    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Q8NDV3", "P42858", "P61158", "P50453", "Q8IX01", "P2196â€¦</span></span>
<span><span class="co">#&gt; $ peptideType   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "unique", "unique", "unique", "non-significant", "uniqueâ€¦</span></span>
<span><span class="co">#&gt; $ proteinType   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "conclusive", "conclusive", "conclusive", "conclusive", â€¦</span></span>
<span><span class="co">#&gt; $ shared        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,â€¦</span></span>
<span><span class="co">#&gt; $ pepScore      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 9.765627e-17, 1.562500e-11, 5.859376e-14, 3.425863e-07, â€¦</span></span>
<span><span class="co">#&gt; $ decoys        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ targets       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 370, 370, 370, 370, 370, 370, 370, 370, 370, 370, 370, 3â€¦</span></span>
<span><span class="co">#&gt; $ pval          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 9.214891e-05, 9.214891e-05, 9.214891e-05, 9.214891e-05, â€¦</span></span>
<span><span class="co">#&gt; $ FDR           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ discPeptides  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4,â€¦</span></span>
<span><span class="co">#&gt; $ totalPeptides <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 8, 10, 6, 8, 12, 7, 7, 6, 35, 9, 69, 50, 9, 11, 6, 8, 15â€¦</span></span>
<span><span class="co">#&gt; $ proteinCount  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 8,â€¦</span></span>
<span><span class="co">#&gt; $ proteinRefs   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Q8NDV3", "P42858", "P61158", "P50453", "Q8IX01", "P2196â€¦</span></span>
<span><span class="co">#&gt; $ proteinMaster <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Q8NDV3", "P42858", "P61158", "P50453", "Q8IX01", "P2196â€¦</span></span>
<span><span class="co">#&gt; $ LP            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4.03551, 4.03551, 4.03551, 4.03551, 4.03551, 4.03551, 4.â€¦</span></span>
<span><span class="co">#&gt; $ qval          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span></code></pre></div>
<p>In a manner equivalent to the protein or gene identification
workflow, we can test the score distribution of the decoy protein
groups:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/plot_rank.html">plot_rank</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="b10prot_files/figure-html/unnamed-chunk-19-1.png" width="672"></p>
<p>And also report a list of groups passing a protein group-level FDR
threshold:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">qval</span> <span class="op">&lt;=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/global_fdr.html">global_fdr</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 Ã— 3</span></span></span>
<span><span class="co">#&gt;   Target Decoy `Global FDR (%)`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>   <span style="text-decoration: underline;">4</span>386    43            0.980</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="refined-fdr">Refined FDR<a class="anchor" aria-label="anchor" href="#refined-fdr"></a>
</h2>
<p>In the identification workflows presented above, we have used the
traditional FDR estimation. However, newer FDR estimation methods based
on a higher-level (e.g., protein) target-decoy competitive approach can
improve sensitivity. This package allows you to compute these FDRs as
long as decoy and target identifications share a common name with an
optional affix (prefix or suffix).</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">refined_genes</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">pep2gene</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Only consider unique (i.e. not shared) peptides</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">shared</span><span class="op">==</span><span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Calculate gene-level scores</span></span>
<span>  <span class="fu"><a href="../reference/lpg.html">lpg</a></span><span class="op">(</span><span class="va">geneRef</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="co"># Calculate refined FDRs using the LPGF score</span></span>
<span>  <span class="fu"><a href="../reference/refined_fdr.html">refined_fdr</a></span><span class="op">(</span><span class="va">geneRef</span>, <span class="va">LPGF</span>, lower_better <span class="op">=</span> <span class="cn">FALSE</span>, affix <span class="op">=</span> <span class="st">"_REVERSED"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">refined_genes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">geneRef</span>, <span class="va">isDecoy</span>, <span class="va">LPGF</span>, <span class="va">FDRn</span>, <span class="va">FDRp</span>, <span class="va">FDRr</span>, <span class="va">to</span>, <span class="va">do</span>, <span class="va">td</span>, <span class="va">tb</span>, <span class="va">db</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 13,552</span></span>
<span><span class="co">#&gt; Columns: 11</span></span>
<span><span class="co">#&gt; $ geneRef <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "A2M", "AARS", "ABAT", "ACAA2", "ACAT1", "ACLY", "ACO1", "ACO2â€¦</span></span>
<span><span class="co">#&gt; $ isDecoy <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE,â€¦</span></span>
<span><span class="co">#&gt; $ LPGF    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 30â€¦</span></span>
<span><span class="co">#&gt; $ FDRn    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ FDRp    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ FDRr    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ to      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 367, 367, 367, 367, 367, 367, 367, 367, 367, 367, 367, 367, 36â€¦</span></span>
<span><span class="co">#&gt; $ do      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ td      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ tb      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span>
<span><span class="co">#&gt; $ db      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,â€¦</span></span></code></pre></div>
<p>And finally, use the refined FDR as a threshold for reported
identifications:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">refined_genes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDRr</span> <span class="op">&lt;=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/global_fdr.html">global_fdr</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 Ã— 3</span></span></span>
<span><span class="co">#&gt;   Target Decoy `Global FDR (%)`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>   <span style="text-decoration: underline;">4</span>282    66             1.54</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Gorka Prieto.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
